prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -60,9 +61,10 @@ static int rx_cm_data(SOCKET fd, struct ofi_ctrl_hdr *hdr,
 		goto out;
 	}
 
-	if (hdr->type != type) {
+	if (hdr->type != ack && (nack != ofi_ctrl_none && hdr->type != nack)) {
```

Comment:
I think we should just check if hdr->type == ofi_ctrl_nack and handle that case.  There shouldn't be a need to pass in a nack value.